testing/coordinator.go

```diff
@@ -26,26 +26,142 @@ var (
 type Coordinator struct {
 	t *testing.T
 
-	Chains map[string]*TestChain
+	CurrentTime time.Time
+	Chains      map[string]*TestChain
 }
 
 // NewCoordinator initializes Coordinator with N TestChain's
 func NewCoordinator(t *testing.T, n int) *Coordinator {
 	chains := make(map[string]*TestChain)
+	coord := &Coordinator{
+		t:           t,
+		CurrentTime: globalStartTime,
+	}
 
 	for i := 0; i < n; i++ {
 		chainID := GetChainID(i)
-		chains[chainID] = NewTestChain(t, chainID)
+		chains[chainID] = NewTestChain(t, coord, chainID)
 	}
-	return &Coordinator{
-		t:      t,
-		Chains: chains,
+	coord.Chains = chains
+
+	return coord
+}
+
+// IncrementTime iterates through all the TestChain's and increments their current header time
+// by 5 seconds.
+//
+// CONTRACT: this function must be called after every Commit on any TestChain.
+func (coord *Coordinator) IncrementTime() {
+	coord.IncrementTimeBy(TimeIncrement)
+}
+
+// IncrementTimeBy iterates through all the TestChain's and increments their current header time
+// by specified time.
+func (coord *Coordinator) IncrementTimeBy(increment time.Duration) {
+	coord.CurrentTime = coord.CurrentTime.Add(increment).UTC()
+	coord.UpdateTime()
+
+}
+
+// UpdateTime updates all clocks for the TestChains to the current global time.
+func (coord *Coordinator) UpdateTime() {
+	for _, chain := range coord.Chains {
+		chain.CurrentHeader.Time = coord.CurrentTime.UTC()
+		chain.App.BeginBlock(abci.RequestBeginBlock{Header: chain.CurrentHeader})
 	}
+
+}
+
+// SetupNew constructs a TM client, connection, and channel on both chains provided. It will
+// fail if any error occurs. The clientID's, TestConnections, and TestChannels are returned
+// for both chains. The channels created are connected to the ibc-transfer application.
+func (coord *Coordinator) SetupNew(path *Path) {
+	coord.SetupConnectionsNew(path)
+
+	// channels can also be referenced through the returned connections
+	coord.CreateMockChannelsNew(path)
```

Comment:
coordinator should look at desired port specified in path and create the channel rather than hardcoding a mock channel.

This will make the function much nicer to use for 3rd party app developers